### PR TITLE
feat(performance-detector-thresholds): Added scroll to setting and uncollapse option functionality.

### PR DESCRIPTION
**Pull request by Abdkhan14**
For project: [Detector Threshold Configuration](https://www.notion.so/sentry/Detector-Threshold-Configuration-f8cb07e7cceb42388cedb09ea05fc116)

- Added scroll to setting + uncollapsing option functionality when navigating to project settings from issue details. 

https://github.com/getsentry/sentry/assets/60121741/2d7a2a47-90de-4028-b7b1-f17151739a1d

### CODE REVIEW

File: static/app/components/forms/jsonForm.tsx [hunk 1]
```diff
@@ -123,7 +127,15 @@ class JsonForm extends Component<Props, State> {
       return null;
     }
 
-    return <FormPanel title={title} fields={fields} {...formPanelProps} />;
+    return (
+      <FormPanel
+        title={title}
+        fields={fields}
+        {...formPanelProps}
```
Comment (k-fish) [inline static/app/components/forms/jsonForm.tsx:134]: Shouldn't this `formPanelProps` already contain the props to be passed to the form panel? Typescript is saying they aren't there because it's casting with `Pick` above I think?

File: static/app/components/forms/formPanel.tsx [hunk 2]
```diff
@@ -55,14 +56,17 @@ function FormPanel({
...
   ...otherProps
 }: Props) {
   const [collapsed, setCollapse] = useState(initiallyCollapsed);
   const handleCollapseToggle = useCallback(() => setCollapse(current => !current), []);
 
   return (
-    <Panel id={typeof title === 'string' ? sanitizeQuerySelector(title) : undefined}>
+    <Panel
+      id={id ?? (typeof title === 'string' ? sanitizeQuerySelector(title) : undefined)}
+    >
```
Comment (k-fish) [inline static/app/components/forms/formPanel.tsx:69]: Instead of `id` can we perhaps use the sanitized title here? Since we already have a standard way of making a panel id?

File: static/app/components/forms/jsonForm.tsx (hunk 1 above)
Comment (Abdkhan14) [reply static/app/components/forms/jsonForm.tsx:134]: Hey @k-fish , so before title fields and now id and initially collapsed comes from formGroup props passed to renderForm because we passed a `forms` prop and not just `fields` to `JsonForm`. [Here](https://github.com/getsentry/sentry/pull/55692/files#diff-9259689032b54efc0d44b1d44a2be5fa972e0d6995481bbc088132ded6ebc252L165). This is different

Comment (Abdkhan14) [reply static/app/components/forms/jsonForm.tsx:134]: I think we can combine formGroup and formPanelProps.

File: static/app/components/forms/formPanel.tsx (hunk 2 above)
Comment (Abdkhan14) [reply static/app/components/forms/formPanel.tsx]: Used title instead.

File: static/app/components/forms/jsonForm.tsx (hunk 1 above)
Comment (k-fish) [reply static/app/components/forms/jsonForm.tsx]: Ah I think I get it now, the initiallyCollapsed state needs to come from the form group itself since we are doing individual collapsing, the formPanel props are just the overarching parent props for initially collapsed. 

We can leave it how you had it previously as an extra param to the function, where you're just passing `initiallyCollapsed` through and it overrides the FormPanel parent prop. Just add a test to assert behaviour when the parent `initiallyCollapsed` is set along side individual form `initiallyCollapsed` as that's maybe one of the edge cases.

**Review by k-fish — APPROVED** [review]
